fastimage.go

```diff
@@ -68,19 +71,21 @@ type decoder struct {
 }
 
 func (f *FastImage) newRequest(url *url.URL, fakeHost string) *http.Request {
+	f.headerMux.Lock()
 	f.header.Set("Referer", url.Scheme+"://"+url.Host)
-
 	req := &http.Request{
 		Method:     "GET",
 		URL:        url,
 		Proto:      "HTTP/1.1",
 		ProtoMajor: 1,
 		ProtoMinor: 1,
-		Header:     *f.header,
+		Header:     f.header.Clone(),
 	}
-	if _, exists := (*f.header)["Host"]; exists {
+	if _, exists := (f.header)["Host"]; exists {
```

Comment:
If you use Clone, you may check Host from req.Header, then the mutex in f.header may not needed.

Reply:
changed , plz review